core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -272,10 +272,23 @@ private NodeInfo asNodeInfo(AdminRow row) {
     }
   }
 
+  /**
+   * @param broadcastRpcAddress this is a parameter only because we already have it when we come
+   *     from {@link #findInPeers(AdminResult, InetSocketAddress)}. Callers that don't already have
+   *     it can use {@link #getBroadcastRpcAddress}.
+   */
   protected DefaultNodeInfo.Builder nodeInfoBuilder(
-      AdminRow row, InetSocketAddress connectAddress) {
-    DefaultNodeInfo.Builder builder = DefaultNodeInfo.builder().withConnectAddress(connectAddress);
+      AdminRow row, InetSocketAddress broadcastRpcAddress) {
+
+    // Deployments that use a custom EndPoint implementation will need their own TopologyMonitor.
+    // One simple approach is to extend this class and override this method.
+    EndPoint endPoint =
+        new DefaultEndPoint(context.getAddressTranslator().translate(broadcastRpcAddress));
```

Comment:
My initial plan was to completely remove `AddressTranslator`, but on second thought I decided to keep it. It fills a sweet spot for simple proxying scenarios where overriding the whole topology monitor would be overkill.